### PR TITLE
make sure error files get built when arangod is built

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Add a depedency so error files get (re)built when arangod is built.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

https://172.16.10.101/view/PR/job/arangodb-matrix-pr/6790/

### CODE REVIEW

File: arangod/CMakeLists.txt [hunk 1]
```diff
@@ -961,6 +961,11 @@ endif ()
 if (USE_JEMALLOC)
   add_dependencies(arangod jemalloc)
 endif ()
+  
+if (USE_MAINTAINER_MODE)
+  add_dependencies(arangod errorfiles)
+  add_dependencies(arangod exitcodefiles)
+endif ()
```
Comment (goedderz) [inline arangod/CMakeLists.txt]: While the error- and exitcodefiles are built now, they might be built after `arango` is built, which contains them. So I think it'd be better to move this to `lib/CMakeLists.txt` and as a dependency to `arango`, not `arangod`.

Comment (jsteemann) [reply arangod/CMakeLists.txt]: True

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/6794/

File: arangod/CMakeLists.txt [hunk 2]
```diff
@@ -961,7 +961,7 @@ endif ()
 if (USE_JEMALLOC)
   add_dependencies(arangod jemalloc)
 endif ()
-
+
```
Comment (goedderz) [inline arangod/CMakeLists.txt]: tz ;)

**Pull request merged**